### PR TITLE
feat: reorganize exports

### PR DESCRIPTION
**Pull request by anitarua**
Addresses https://github.com/momentohq/client-sdk-rust/issues/194 

* Removed the proto exports from the top-level `lib.rs`; re-exported out of the `response` namespace instead.
* Renamed `basic` folder to `scalar` to be more consistent with other SDKs' naming schemes.
* Renamed the `requests` folder to `cache` and the `cache` folder to `requests`.
* Re-exported stuff individually out of `requests` (thus hiding `requests` from the import path and putting it under the `cache` namespace instead).

### CODE REVIEW

File: example/src/bin/readme.rs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 use momento::config::configurations::laptop;
-use momento::requests::cache::basic::get::Get;
+use momento::cache::{Get, SortOrder, SortedSetFetch};
```
Comment (anitarua) [inline example/src/bin/readme.rs:2]: simplified import path

File: src/cache/mod.rs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// Make module "requests" private and re-export its contents so users can
+// import using "momento::cache::{Get, Set, ...}"
+mod requests;
```
Comment (anitarua) [inline src/cache/mod.rs:3]: Re-exporting as per https://doc.rust-lang.org/rustdoc/write-documentation/re-exports.html

File: src/lib.rs [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
+*/
 pub mod auth;
+pub mod cache;
 /// Configuration structs for the Momento clients.
 pub mod config;
+pub mod errors;
 pub mod preview;
-pub mod requests;
 pub mod response;
+/*************************************************************/
```
Comment (anitarua) [inline src/lib.rs:12]: are we okay with these namespaces?

hadn't rearranged anything related to topics yet, but should we rename it from "preview" to just "topics"?

File: example/src/bin/readme.rs [hunk 4]
```diff
@@ -17,6 +17,13 @@ pub async fn main() -> Result<(), MomentoError> {
 
     cache_client.create_cache(CACHE_NAME.to_string()).await?;
 
+    cache_client.flush_cache(CACHE_NAME).await?;
+
+    match cache_client.list_caches().await {
```
Comment (cprice404) [inline example/src/bin/readme.rs:22]: FYI - this readme.ts file gets sucked into the README template, so we want to keep it super short. If you want to make a copy of this file and call it `cache_operations.rs` or something, and keep the `readme.ts` file short like it was before, that would work!

File: example/src/bin/readme.rs (hunk 1 above)
Comment (cprice404) [reply example/src/bin/readme.rs:2]: ❤️

File: src/cache/mod.rs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+pub use requests::delete_cache::{DeleteCache, DeleteCacheRequest};
+pub use requests::flush_cache::{FlushCache, FlushCacheRequest};
+pub use requests::list_caches::{ListCaches, ListCachesRequest};
+
+pub use requests::scalar::get::{Get, GetRequest, GetValue};
+pub use requests::scalar::set::{Set, SetRequest};
+
+pub use requests::set::set_add_elements::{SetAddElements, SetAddElementsRequest};
+
+pub use requests::sorted_set::sorted_set_fetch_by_rank::{SortOrder, SortedSetFetchByRankRequest};
```
Comment (cprice404) [inline src/cache/mod.rs:17]: if this SortOrder is really specific to SortedSets we might need to disambiguate that in some way, like name it SortedSetSortOrder or SortedSet::SortOrder or something. If it's not specific to sorted sets then we could move its definition to a more generic location. Just a nit; this is not really something specific to this PR, can sort it out later (pun intended)

File: src/lib.rs (hunk 3 above)
Comment (cprice404) [reply src/lib.rs:12]: good callout, ty

let's make another pass to deal with topics after we are happy with cache stuff.

response namespace will go away once we remove the old client, i think.

we should check out auth and config and see if we want to re-export stuff from them either directly under `momento`, or into `cache` (e.g. for configuration, which is probably going to be cache-specific). Don't feel obligated to hit those in this PR, but maybe make a ticket for it.

File: src/lib.rs [hunk 6]
```diff
@@ -13,19 +19,12 @@ mod grpc;
...
 pub use crate::cache_client::CacheClient;
 
 pub type MomentoResult<T> = Result<T, MomentoError>;
-
-pub mod sorted_set {
-    pub use momento_protos::cache_client::sorted_set_fetch_request::{Order, Range};
-    pub use momento_protos::cache_client::sorted_set_fetch_response::found::Elements;
-    pub use momento_protos::cache_client::sorted_set_fetch_response::SortedSet;
-    pub use momento_protos::cache_client::SortedSetElement;
-}
```
Comment (cprice404) [inline src/lib.rs:31]: :ty:

**Review by cprice404 — COMMENTED** [review]
left a few comments but I don't think any of them are blockers except for keeping `readme.ts` short. looking good!

File: example/src/bin/readme.rs (hunk 4 above)
Comment (anitarua) [reply example/src/bin/readme.rs:22]: was planning on reverting the changes in `readme.ts` before taking this PR out of draft, just committed it for review purposes for now 👍

Comment (cprice404) [reply example/src/bin/readme.rs:22]: gotcha. very helpful for review ty

File: src/lib.rs (hunk 3 above)
Comment (anitarua) [reply src/lib.rs:12]: captured in ticket https://github.com/momentohq/client-sdk-rust/issues/222

File: src/cache/mod.rs (hunk 5 above)
Comment (anitarua) [reply src/cache/mod.rs:17]: Looks like it's called SortedSetOrder in the other SDKs so will follow suit here

File: src/response/mod.rs [hunk 7]
```diff
@@ -29,6 +29,13 @@ pub use self::generate_api_token_response::*;
 pub use self::list_cache_response::*;
 pub use self::list_signing_keys_response::*;
 
+pub mod simple_cache_client_sorted_set {
+    pub use momento_protos::cache_client::sorted_set_fetch_request::{Order, Range};
+    pub use momento_protos::cache_client::sorted_set_fetch_response::found::Elements;
+    pub use momento_protos::cache_client::sorted_set_fetch_response::SortedSet;
+    pub use momento_protos::cache_client::SortedSetElement;
+}
+
```
Comment (anitarua) [inline src/response/mod.rs:38]: these were still used in the simple cache client doctests so I opted to re-export them from here to make it easier to remove these once we remove the simple cache client entirely

**Review by cprice404 — DISMISSED** [review]
yeah

**Review by cprice404 — APPROVED** [review]

**Pull request merged**